src/main/java/org/gwtbootstrap3/extras/toggleswitch/client/ui/base/ToggleSwitchBase.java

```diff
@@ -206,21 +209,59 @@ public void setValue(final Boolean value) {
         setValue(value, false);
     }
 
+    /**
+     * Called when this widget is attached to the DOM.
+     * 
+     * Here we want to make sure the state of the toggle switch hasn't changed before it was 
+     * attached or re-attached to the DOM.  If the value has changed update the toggle switch 
+     * via {@link #switchState(Element, boolean, boolean)}.
+     */
+    @Override
+    public void onAttach() {
```

Comment:
Why do not call ONLY once switchState() ? It ensures the value synchronization between toggleSwitch and input. If they are different, you don't need to call it twice.
````
@Override
public void onAttach() {
  super.onAttach();
  switchState(getElement(), getValue(), true);
}
```